core/src/main/scala/kafka/consumer/PartitionAssignor.scala

```diff
@@ -77,18 +72,11 @@ class RoundRobinAssignor() extends PartitionAssignor with Logging {
       new Pool[String, mutable.Map[TopicAndPartition, ConsumerThreadId]](Some(valueFactory))
 
     if (ctx.consumersForTopic.size > 0) {
-      // check conditions (a) and (b)
-      val (headTopic, headThreadIdSet) = (ctx.consumersForTopic.head._1, ctx.consumersForTopic.head._2.toSet)
-      ctx.consumersForTopic.foreach { case (topic, threadIds) =>
-        val threadIdSet = threadIds.toSet
-        require(threadIdSet == headThreadIdSet,
-          "Round-robin assignment is allowed only if all consumers in the group subscribe to the same topics, " +
-            "AND if the stream counts across topics are identical for a given consumer instance.\n" +
-            "Topic %s has the following available consumer streams: %s\n".format(topic, threadIdSet) +
-            "Topic %s has the following available consumer streams: %s\n".format(headTopic, headThreadIdSet))
+      val allThreadIds = ctx.consumersForTopic.flatMap { case (topic, threadIds) =>
```

Comment:
A `ConsumerThreadId` can be interested in multiple topics, so `allThreadIds` could have duplicate `ConsumerThreadId`s. Do we want to deduplicate?